components/support/nimbus-fml/src/client.rs

```diff
@@ -0,0 +1,237 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+#![allow(unused)]
+
+use crate::error::ClientError::{
+    InvalidFeatureConfig, InvalidFeatureId, InvalidFeatureValue, JsonMergeError,
+};
+use crate::error::FMLError::ClientError;
+use crate::{
+    error::{FMLError, Result},
+    intermediate_representation::{FeatureManifest, TypeRef},
+    parser::Parser,
+    util::loaders::FileLoader,
+};
+use serde::Deserialize;
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct FeatureConfig {
+    pub feature_id: String,
+    pub value: serde_json::Value,
+}
+
+pub struct MergedJsonWithErrors {
+    pub json: String,
+    pub errors: Vec<FMLError>,
+}
```

Comment:
Is this struct ever going to be both `json` and `errors`?

Would it be better to use an enum, or even just a `Result` enum?

Reply:
Yep! It will always return with the `json`, which will be at least the default JSON, but would also include any valid feature configs merged in. `errors` is a list of errors found while merging. Those feature configs don't get merged in, and the `errors` list is a way for us to note why and report it up to Sentry.